Documentation/Common-Gotchas.md

```diff
@@ -0,0 +1,232 @@
+# Common Gotchas
+
+There are some common errors that arise when trying to use Argo. Many are due to
+ the fact that Argo is so fundamentally different from how you might be used to
+ decoding JSON, and others are due to limitations of Swift. This document aims
+ to address some of those situations and set you on the right track!
+
+## `curry` limitations
+
+It is very important to fully curry your model’s `init` in order to use Argo.
+Because of this, `Argo` has a dependency on
```

Comment:
Argo isn't actually dependent on Curry. Perhaps say that it is much easier to use with the curry library